commit 82d3d4610c3036c37b025dfd954d63b2dc220459 (refs/changes/46/2146/6)
Author: Sean Chittenden <sean@chittenden.org>
Date:   2017-06-27T11:22:41-07:00 (2 years, 3 months ago)
    
    MANTA-3323: initdb(1) with explicit locale and encoding
    
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/postgresMgr.js b/lib/postgresMgr.js
index 620fc9d..4efe456 100644
--- a/lib/postgresMgr.js
+++ b/lib/postgresMgr.js
@@ -29,6 +29,7 @@ var fs = require('fs');
 var pg = require('pg');
 var Client = pg.Client;
 var shelljs = require('shelljs');
+var mod_forkexec = require('forkexec');
 var spawn = require('child_process').spawn;
 var exec = require('child_process').exec;
 var once = require('once');
@@ -1497,17 +1498,14 @@ PostgresMgr.prototype._initDb = function (callback) {
 
         },
         function _initDb(_, cb) {
-            var cmd = 'sudo -u ' + self._dbUser + ' ' +
-                self._pgInitDbPath + ' -D ' + self._dataDir;
-            log.info({cmd: cmd}, 'PostgresMgr.initDb: initializing db');
-            exec(cmd, function (err, stdout, stderr) {
+            var args = [ 'sudo', '-u', self._dbUser,
+                         self._pgInitDbPath, '--encoding=UTF-8', '--locale=C',
+                         '-D', self._dataDir ];
+            log.info({cmd: 'initdb', argv: args},
+                     'PostgresMgr.initDb: initializing db');
+            mod_forkexec.forkExecWait({ argv: args }, function (err, info) {
                 // ignore errors since the db could already be initialized
-                log.info({
-                    err: err,
-                    stdout: stdout,
-                    stderr: stderr,
-                    dataDir: self._dataDir
-                }, 'PostgresMgr.initDb: initdb returned');
+                log.info(info, 'PostgresMgr.initDb: initdb returned');
 
                 shelljs.cp('-f', self._hbaConf, self._dataDir + '/pg_hba.conf');
                 log.info({
